x-pack/distributions/internal/observabilitySRE/plugin/logstash-integration-fips_validation/lib/logstash/fips_validation.rb

```diff
@@ -0,0 +1,99 @@
+
+require "logstash/environment"
+
+require "logstash/plugins/registry"
+
+module LogStash
+  class FipsValidation < LogStash::UniversalPlugin
+
+    include LogStash::Util::Loggable
+
+    require 'java'
+    java_import org.jruby.util.SafePropertyAccessor
+
+    def register_hooks(hooks)
+      logger.debug("registering hooks")
+      require 'logstash/runner'
+      hooks.register_hooks(LogStash::Runner, self)
+    end
+
+    def before_bootstrap_checks(runner)
```

Comment:
this could be _any_ of the hooks that are fired by [the `LogStash::Runner`'s `LogStash::EventDispatcher`](https://github.com/elastic/logstash/blob/feature/fedramp-high-8.x/logstash-core/lib/logstash/runner.rb#L359), but `before_bootstrap_checks` is the first one it fires.